### ISSUE TITLE
Imagebind for commercial use

### ISSUE THREAD

**Issue by abhimanyu891998** [facebookresearch/ImageBind#90]
If one only intends to use audio,video and text embeddings from imageBind for a project intended for commercial use, would Meta AI allow for such a usecase? I see the model is available for non-commercial use only, but was just wondering given that video and text is essentially CLIP in imageBind, or, will Meta ever release this for commercial use?

**Comment by onexdata** [comment]
Since you didn't get an answer for a while, here is my legal analysis (although I'm not a lawyer and this isn't legal advice):

**TL/DR; The vectors ImageBind generated, ahead-of-time, and using them in a commercial product to do NN vector comparisons are likely to be considered facts and not enforceable in any license, however, using ImageBind for something like real-time search (turning input to vector, in real-time, when a search is requested, in a commercial product) would almost certainly violate the NC-SA license.**


1. You cannot copyright an actual fact or mathematical representation of something, regardless of effort to produce or how it was created.

The U.S. Supreme Court affirmed this concept in the landmark case **_"Feist Publications, Inc. v. Rural Telephone Service Co."_**, stating that "facts are not copyrightable" and specifically including any mathematical representation. The Court emphasized that copyright protection extends only to the author's original expression, not to any facts that may be derived themselves, regardless of how much effort went into discovering or compiling them.)

2. As of 1/22/2025, it looks like nobody has ever attempted to actually sue anyone yet over using NC-SA embeddings in commercial products.

3. According to Berkely Tech Law Journal, it's unlikely anyone would win a lawsuit involving data mining or representing anything that may be copyrighted, such as in a vector format - https://btlj.org/wp-content/uploads/2023/02/0003-36-4Quang.pdf

5. Although not exactly the same, the NYT, NY Daily News and the Center for Investigative Reporting are all suing OpenAI, claiming their articles are copyrighted, and OpenAI's model weights, although much more detailed than embeddings and not quite embeddings, are a copyright violation. This is currently awaiting a motion for summary judgement. If this lawsuit fails, it is very likely embeddings, which are less than model weights, would be enforceable, if they win against OpenAI, then it will at least not close the door. https://www.npr.org/2025/01/14/nx-s1-5258952/new-york-times-openai-microsoft.

**HOWEVER**

The Feist ruling would likely protect your ability to USE the vectors themselves commercially - the argument being they're essentially mathematical facts about the inputs, regardless of how they were derived. Just like you can't copyright the phone numbers in a phone book, you can't copyright vector representations of data properties.
 
However, the actual process of GENERATING those vectors in real-time requires running the ImageBind model itself. This would constitute using the model (a creative work) in a commercial context, which would **_almost certainly_** violate the NC-SA license.
